### ISSUE TITLE
Terraform doesn't update in place

### ISSUE THREAD

**Issue by oscarsjlh** [hashicorp/terraform-provider-google#12436]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
* Please do not leave _+1_ or _me too_ comments, they generate extra noise for issue followers and do not help prioritize the request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.
* If an issue is assigned to the `modular-magician` user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If an issue is assigned to a user, that user is claiming responsibility for the issue. If an issue is assigned to `hashibot`, a community member has claimed the issue already.

### Terraform Version
*Terraform v1.2.7

* on linux_amd64

* provider registry.terraform.io/hashicorp/google v4.0.0

* provider registry.terraform.io/hashicorp/random v3.3.2 

### Affected Resource(s)

* google_XXXXX

### Terraform Configuration Files

```
resource "google_compute_firewall" "allow_controller" {
allow {
   
 ports    = [22, 8080, 8443]
    protocol = "tcp"
  }
  allow {
    ports    = [3478]
    protocol = "udp"
  }
  direction     = "INGRESS"
  name          = "allow-controller"
  network       = var.network 
  priority      = 1000
  project       = var.project_network
  source_ranges = ["${var.office_ip}", "${var.vpn_uk}", "${var.vpn_us}", "${var.vpn_fr}",  "${var.office_ip_us}"]
  target_tags   = ["controller"]
}`
```

# Copy-paste your Terraform configurations here.
#
# For large Terraform configs, please use a service like Dropbox and share a link to the ZIP file.
# For security, you can also encrypt the files using our GPG public key:
#    https://www.hashicorp.com/security
#
# If reproducing the bug involves modifying the config file (e.g., apply a config,
# change a value, apply the config again, see the bug), then please include both:

# * the version of the config before the change, and

```
resource "google_compute_firewall" "allow_controller" {
 
allow {
   
 ports    = [22, 8080, 8443]
    protocol = "tcp"
  }
  allow {
    ports    = [3478]
    protocol = "udp"
  }
  direction     = "INGRESS"
  name          = "allow-controller"
  network       = var.network 
  priority      = 1000
  project       = var.project_network
  source_ranges = ["${var.office_ip}", "${var.vpn_uk}", "${var.vpn_us}", "${var.vpn_fr}",  ]
  target_tags   = ["controller"]
}`
```
# * the version of the config after the change.

```
resource "google_compute_firewall" "allow_controller" {
 
allow {
   
 ports    = [22, 8080, 8443]
    protocol = "tcp"
  }
  allow {
    ports    = [3478]
    protocol = "udp"
  }
  direction     = "INGRESS"
  name          = "allow-controller"
  network       = var.network 
  priority      = 1000
  project       = var.project_network
  source_ranges = ["${var.office_ip}", "${var.vpn_uk}", "${var.vpn_us}", "${var.vpn_fr}",  "${var.office_ip_us}"]
  target_tags   = ["controller"]
}
```

### Expected Behavior

If I add one more vlalue on the source_ranges it should update_inplace and just add that value to the firewall rule 

### Actual Behavior

 Terraform removes all previous values in source_ranges and just keeps the new one, if I plan again it gives me the ones it removed before. And is stuck in this loop

### Steps to Reproduce

 Please list the steps required to reproduce the issue
Create the variable 
Add the variable to the source_ranges [
Terraform init
Terraform plan
Terraform apply 

### Important Factoids
This are imported firewall rules

### References

* #0000

**Comment by edwardmedia** [comment]
@oscarsjlh adding / removing a range in the `source_ranges` seems working fine with my test. Mind share your debug log so I can take a close look?

**Comment by oscarsjlh** [comment]
> 
Hi sure should I try to remove any of my public ip in the logs or is it fine?

**Comment by oscarsjlh** [comment]
Yeah sure should I try to block any public ip there might be there or is it fine?

**Comment by oscarsjlh** [comment]
@edwardmedia Yes terraform runs without any errors, and It keeps the last value added in "Source ranges", but if I plan and apply again it wants to put the old ones back, and removes the new one, and if I plan again it wants the new one and removes the old ones. And it's stuck on that loop

I've tried forgetting all the resources, and importing them again, deleting the module containing the resource, and writing it again and nothing seems to solve the issue.

The tf state is saved in a bucket in GCP if that is of any help.

**Comment by edwardmedia** [comment]
@oscarsjlh I see what you meant now. This bug has been fixed in [v4.1.0](https://github.com/hashicorp/terraform-provider-google/releases/tag/v4.1.0). Can you try the latest provider to see if it works as expected?

**Comment by oscarsjlh** [comment]
@edwardmedia That did the trick, I feel kind of dumb to not think of that. Thanks very much
